### PR TITLE
Add translations documentation

### PR DESCRIPTION
**Pull request by JulienPradet**

### CODE REVIEW

**Comment by JulienPradet** [comment]
I didn't talk about magento's translations yet.

I don't know where it makes the most sense to document this.

File: source/docs/advanced/theme/translations.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+Big e-commerce websites often need to translate their content in several languages to make sure that they reach a broader range of customers.
+
+In this documentation, we will see how this works for static content in your application.
+
+<blockquote class="note">
+If you want to configure your application to support multiple languages, please refer to [Configure multiple stores](/docs/advanced/production-ready/multistore.html).
+</blockquote>
+
+Front-Commerce manages your translations by using [`react-intl`](https://github.com/yahoo/react-intl), a standard in the React ecosystem. This library works by transforming your static values in React Components. These components will then fetch your translations and display them in your application.
```
Comment (real34) [inline source/docs/advanced/theme/translations.md]:
```suggestion
Front-Commerce manages your translations by using [`react-intl`](https://github.com/yahoo/react-intl), a standard library in the React ecosystem. This library works by transforming your static values in React Components. These components will then fetch your translations and display them in your application.
```

File: source/docs/advanced/theme/translations.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
+---
+id: translations
+title: Translate your application
+---
+
+Big e-commerce websites often need to translate their content in several languages to make sure that they reach a broader range of customers.
```
Comment (real34) [inline source/docs/advanced/theme/translations.md]: I am not a fan of this sentence, but it’s difficult to point out what makes me feel like that exactly. Here are suggestions to express ideas, we could discuss it to find a correct one if you want.

```suggestion
International e-commerce websites often need to translate their UI in several languages to make sure that they reach a broader range of customers.
```

or

```suggestion
Global e-commerce websites often need to allow users to choose their preferred language to make sure that they reach a broader range of customers.
```

File: source/docs/advanced/theme/translations.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+In this documentation, we will see how this works for static content in your application.
+
+<blockquote class="note">
+If you want to configure your application to support multiple languages, please refer to [Configure multiple stores](/docs/advanced/production-ready/multistore.html).
```
Comment (real34) [inline source/docs/advanced/theme/translations.md]:
```suggestion
Often you may want to change dynamic content (CMS pages, product descriptions…) in addition to translating the UI. If you want to configure your application to support multiple languages, please refer to [Configure multiple stores](/docs/advanced/production-ready/multistore.html).
```

File: source/docs/advanced/theme/translations.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+- <span>{value.amount} €</span>
++ import { FormattedNumber } from "react-intl";
++ <FormattedNumber
++   value={value.amount}
++   style="currency"
++   currency={value.currency}
++ />
+```
+    <blockquote class="note">
+    However in this particular we have abstracted this in Front-Commerce with the [`theme/components/atoms/Price`](https://gitlab.com/front-commerce/front-commerce/tree/develop/src/web/theme/components/atoms/Typography/Price) component.
```
Comment (real34) [inline source/docs/advanced/theme/translations.md]:
```suggestion
    However, for this particular use case we have abstracted this in Front-Commerce with the [`theme/components/atoms/Price`](https://gitlab.com/front-commerce/front-commerce/tree/develop/src/web/theme/components/atoms/Typography/Price) component (and its variants: `ProductPrice`, `PriceOrFree`). We recommend you to use it instead, so you could benefit from better integration with its related GraphQL Type.
```

File: source/docs/advanced/theme/translations.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+These components will be enough in most cases. However, the issue with these is that it wraps your string within a React Element, which may be troublesome when you actually need to handle the string itself.
+
+For instance, this is the case when you want to add some label attributes to your DOM elements.
+
+```jsx
+<span class="icon" aria-label="Icon title displayed for screen readers" />
+```
+
+Fortunatly, this is correctly handled by react-intl if you use [`defineMessages`](https://github.com/yahoo/react-intl/wiki/API#definemessages) combined with [`injectIntl`](https://github.com/yahoo/react-intl/wiki/API#injection-api).
```
Comment (real34) [inline source/docs/advanced/theme/translations.md]:
```suggestion
Fortunately, this is correctly handled by react-intl if you use [`defineMessages`](https://github.com/yahoo/react-intl/wiki/API#definemessages) combined with [`injectIntl`](https://github.com/yahoo/react-intl/wiki/API#injection-api).
```

File: source/docs/advanced/theme/translations.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+<blockquote class="wip">
+Please refer to [react-intl documentation](https://github.com/yahoo/react-intl/wiki/API#injection-api) for more information about these two methods. We plan to add an example here in the future but didn't have time to document it yet.
+</blockquote>
+
+## Setup your translations what's in your components
```
Comment (real34) [inline source/docs/advanced/theme/translations.md]: I really do not see what you wanted to write here… but there definitely seems to be a mistake!

File: source/docs/advanced/theme/translations.md [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+
+Now that you have defined what should be translated in your application, you actually need to translate your application. This is doable in two steps:
```
Comment (real34) [inline source/docs/advanced/theme/translations.md]:
```suggestion
Now that you have defined what should be translated in your application, you actually need to translate your application. This is a two-step process:
```

File: source/docs/advanced/theme/translations.md [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+
+This task can be daunting at first, but it will make your life easier in the future when you will need to know which string was not translated in a recent update.
+
+## About dynamic content
+
+`react-intl` lets you translate your static content. But what about dynamic content? Things like product title and description, CMS pages, etc.
+
+This is done on the GraphQL side. The principle is that when a user is connected to Front-Commerce, they will be assigned a `storeViewCode` in their session (see [Configure multiple stores](/docs/advanced/production-ready/multistore.html) for more details).
+
+This code will then be used by your GraphQL loaders to retreive the correct content from your backend. Learn more about [GraphQL loaders](/docs/advanced/graphql/slim-down-resolvers-with-loaders.html).
```
Comment (real34) [inline source/docs/advanced/theme/translations.md]:
```suggestion
This code will then be used by your GraphQL loaders to retrieve the correct content from your backend. Learn more about [GraphQL loaders](/docs/advanced/graphql/slim-down-resolvers-with-loaders.html).
```

File: source/docs/advanced/theme/translations.md (hunk 3 above)
Comment (real34) [reply source/docs/advanced/theme/translations.md]: I saw it was also mentioned in the last section, so feel free to discard this suggestion if you’d like.

**Pull request merged**